### DISCUSSION TITLE
PIP-28: Prevent Node Shutdowns for Validators in Committee

### DISCUSSION THREAD

**Discussion by ja7ad** [pactus-project/PIPs#163] [PIPs]
PIP-28: [Prevent Node Shutdowns for Validators in Committee](https://pips.pactus.org/PIPs/pip-28)

**Comment by b00f** [comment]
I don't agree with this PIP. What are the chances that a validator in the committee wants to restart? It's really low, and for those rare cases, we're going to add code to the project .

Moreover, if a validator in the committee restarts, nothing happens except for one absentee in the certificate. 

Users can check if they are in the committee and choose not to restart their node if they want.

**Comment by kehiy** [comment]
In my opinion about this PIP, as I said [before](https://github.com/pactus-project/PIPs/pull/154#issuecomment-2211734678):

We can't count a software improvement as a PIP. This feature is useful and greatly helps users, but I believe it's not a PIP.

We must be able to refer to PIPs as a reference of **Protocol** later.
In most chains, people follow them to write new protocol implementations. or even use them on other projects.

Considering a software improvement as a PIP will make a huge amount of PIPs in the future all about software and all of them have small descriptions and specifications...

Now, think you are a developer exploring pactus to write a new implementation or make a new tool. what do you think?

In my opinion, it's not a good experience. It's better to find a proper line between software and protocol improvements.

**Reply by b00f** [reply]
Check [PIP-1](https://pips.pactus.org/PIPs/pip-1) and propose changes there. Any changes related to PIPs or their workflow should be reflected there. If we want to filter out the PIPs, we have to regulate it there.

**Reply by kehiy** [reply]
@b00f good idea. I'll make changed there and make a PR.
